src/plugin.ts

```diff
@@ -327,7 +327,11 @@ export class CycloneDxWebpackPlugin {
     if (component === undefined) { return }
     if (
       typeof this.rootComponentBuildSystem === 'string' &&
-      this.rootComponentBuildSystem.length > 0
+      this.rootComponentBuildSystem.length > 0 &&
+      !iterableSome(
+        component.externalReferences,
+        ref => ref.type === CDX.Enums.ExternalReferenceType.BuildSystem
```

Comment:
Don't add if it is the exact same URL:
```suggestion
        ref => ref.type === CDX.Enums.ExternalReferenceType.BuildSystem &&
          ref.url === this.rootComponentBuildSystem
```

Reply:
debatable. see https://github.com/CycloneDX/cyclonedx-webpack-plugin/pull/1355#issuecomment-2619521529

Reply:
debate done. all fine.